dbt_dry_run/node_runner/incremental_runner.py

```diff
@@ -175,6 +175,38 @@ def _get_full_refresh_config(self, node: Node) -> bool:
             return node.config.full_refresh
         return flags.FULL_REFRESH
 
+    def _is_time_ingestion_partitioned(self, node: Node) -> bool:
+        if node.config.partition_by:
+            if node.config.partition_by.time_ingestion_partitioning is True:
+                return True
+        return False
+
+    def _replace_partition_with_time_ingestion_column(
+        self, dry_run_result: DryRunResult
+    ) -> DryRunResult:
+        if not dry_run_result.table:
+            return dry_run_result
+
+        if not dry_run_result.node.config.partition_by:
+            return dry_run_result
+
+        existing_partition_field = dry_run_result.node.config.partition_by.field
+
+        new_partition_field = TableField(
+            name="_PARTITIONTIME",
+            type=BigQueryFieldType.TIMESTAMP,
+            mode=BigQueryFieldMode.NULLABLE,
+        )
+
+        final_fields = [
```

Comment:
How come we are removing the field from the predicted schema? If time ingestion partitioning is enabled then I can see how `_PARTITIONTIME` gets added to the predicted schema but I don't see why you would remove the partition_by field from the predicted schema?

Reply:
Assuming `time_ingestion_partitioning` is set to true, I thought that there wouldn't be any queries that select the `partition_by.field` since it essentially won't be a field in the real table. I put it back to the predicted schema.